kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,231 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm.configuration;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.freedesktop.dbus.errors.NotSupported;
+import org.junit.Test;
+
+public class NetworkPropertiesTest {
+
+	NetworkProperties netProps;
+	Map<String, Object> properties;
+	Optional<?> optResult;
+	String stringResult;
+	List<String> stringListResult;
+	Map<String, Object> resultMap;
+
+	@Test(expected = NullPointerException.class)
+	public void shouldFailWhenNull() {
+		givenNetworkPropertiesBuiltWith(null);
+		givenNetworkPropsIsCreated();
+	}
```

Comment:
Additionally this is not completely adherent to our Gerkhin-style testing guidelines.

It should be something like:
```suggestion
	public void shouldFailWhenNull() {
		givenNetworkPropertiesBuiltWith(null);
		givenNetworkPropsIsCreated();
		thenNullPointerExceptionIsThrown()
	}
```

Look ad what we did [here](https://github.com/eclipse/kura/blob/develop/kura/test/org.eclipse.kura.ai.triton.server.test/src/test/java/org/eclipse/kura/ai/triton/server/TritonServerEncryptionUtilsTest.java) with the `thenAnExceptionOccurred`... it would be nice if you checked for what kind of exception is thrown though.